packages/core/src/components/icon/icon.tsx

```diff
@@ -71,29 +74,27 @@ export class Icon extends React.PureComponent<IIconProps & React.DOMAttributes<H
     public static readonly SIZE_LARGE = 20;
 
     public render() {
+        const { icon } = this.props;
+        if (icon == null) {
+            return null;
+        } else if (typeof icon !== "string") {
+            return icon;
+        }
+
         const {
             className,
             color,
-            icon,
             iconSize = Icon.SIZE_STANDARD,
             intent,
             title = icon,
             tagName: TagName = "span",
             ...htmlprops
         } = this.props;
 
-        if (icon == null) {
-            return null;
-        } else if (typeof icon !== "string") {
-            return icon;
-        }
-
         // choose which pixel grid is most appropriate for given icon size
         const pixelGridSize = iconSize >= Icon.SIZE_LARGE ? Icon.SIZE_LARGE : Icon.SIZE_STANDARD;
+        // render path elements, or nothing if icon name is unknown.
         const paths = this.renderSvgPaths(pixelGridSize, icon);
-        if (paths == null) {
-            return null;
-        }
```

Comment:
so now it will always have an empty `<svg>` instead of nothing?

Reply:
if you say `<Icon icon="blah" />` then yes. `<Icon icon={null} />` renders `null`.